Access-Control: Document changes to the provisioner (#35548)
grafana/grafana: docs/sources/enterprise/access-control/provisioning.md

### Delete roles 

> To delete a role, you can add a list of roles under the `deleteRoles` section in the configuration file. 

Reviewer: ```suggestion
To delete a role, add a list of roles under the `deleteRoles` section in the configuration file. 
```

> > Deletion of roles listed under the `deleteRoles` section happens **before** saving roles listed under the `roles` section.

Reviewer: ```suggestion
> **Note:** Any role in the `deleteRoles` section is deleted before any role in the `roles` section is saved.
```

To assign roles to built-in roles, add said built-in roles to the `builtInRoles` section of your roles. To remove a specific assignment, remove it from the list.

> > Assignments are updated if the version of the role is greater or equal to the one stored internally. You don't need to increment the version number of the role to update its assignments.

Reviewer: ```suggestion
> **Note:** Assignments are updated if the version of the role is greater or equal to the one stored internally. You don’t need to increment the version number of the role to update its assignments.
```

> > Assignments are updated if the version of the role is greater or equal to the one stored in database. You don't need to increment the version number of the role to update its assignments.

Author: ```suggestion
> Assignments are updated if the version of the role is greater or equal to the one stored internally. You don't need to increment the version number of the role to update its assignments.
```
